source/adaptive_load/session_spec_proto_helper_impl.cc

```diff
@@ -0,0 +1,119 @@
+#include "adaptive_load/session_spec_proto_helper_impl.h"
+
+#include "nighthawk/adaptive_load/metrics_plugin.h"
+#include "nighthawk/adaptive_load/step_controller.h"
+
+#include "api/adaptive_load/adaptive_load.pb.h"
+#include "api/adaptive_load/metric_spec.pb.h"
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/string_view.h"
+#include "adaptive_load/metrics_plugin_impl.h"
+#include "adaptive_load/plugin_loader.h"
+
+namespace Nighthawk {
+
+nighthawk::adaptive_load::AdaptiveLoadSessionSpec
+AdaptiveLoadSessionSpecProtoHelperImpl::SetSessionSpecDefaults(
+    nighthawk::adaptive_load::AdaptiveLoadSessionSpec spec) {
+  if (!spec.nighthawk_traffic_template().has_open_loop()) {
+    spec.mutable_nighthawk_traffic_template()->mutable_open_loop()->set_value(true);
+  }
+  if (!spec.has_measuring_period()) {
+    spec.mutable_measuring_period()->set_seconds(10);
```

Comment:
Should we have these as consts at the top so that they're easier to modify, rather than being numbers inside the method itself? E.g. set_seconds(DEFAULT_MEASURING_PERIOD)

Reply:
Just to chime in, I think this is a matter of opinion, but in the past, I think I've seen mum4k argue against constants for this purpose, because they're only used once, and potentially create misdirection when trying to figure out what the defaults are.

This function is already properly abstracted to be a SetDefaults method, so it's relatively separated from the rest of the code, and easy to find how it works.

Reply:
Ok. LGTM.